### PR TITLE
Adding end-to-end example for L7 ILB

### PR DESCRIPTION
**Pull request by betsy-lichtenberg**
This example creates an internal HTTP load balancer with a Compute Engine backend. The backend is created via managed instance group. The front end uses HTTP, not HTTPS.

```release-note:none
```

### CODE REVIEW

**Review by c2thorn — CHANGES_REQUESTED** [review]
Hi @betsy-lichtenberg,

Something I didn't catch before - the new template file will need to be in `mmv1/templates/terraform/examples`, not just `mmv1/templates/terraform`. The tests aren't running yet since MM thinks the file does not exist.

Also for PRs that only affect examples, we can usually leave them out of the changelog entry. That means we can use "release-note:none" as the tag and leave entry empty. I'll go ahead and edit that now

**Comment by betsy-lichtenberg** [comment]
> Hi @betsy-lichtenberg,
> 
> Something I didn't catch before - the new template file will need to be in `mmv1/templates/terraform/examples`, not just `mmv1/templates/terraform`. The tests aren't running yet since MM thinks the file does not exist.
> 

Done, but I'm not clear on whether I should remove the new template file `mmv1/templates/terraform`. Please confirm.

**Comment by c2thorn** [comment]
> Done, but I'm not clear on whether I should remove the new template file `mmv1/templates/terraform`. Please confirm.

Yep, you can go ahead and remove the `mmv1/templates/terraform` version, it won't do anything in that location.

File: mmv1/products/compute/terraform.yaml [hunk 1]
```diff
@@ -702,6 +702,24 @@ overrides: !ruby/object:Overrides::ResourceOverrides
...
+          backend_subnet_name: "l7-ilb-subnet"
+          forwarding_rule_name: "l7-ilb-forwarding-rule"
+          target_http_proxy_name: "l7-ilb-target-http-proxy"
+          regional_url_map_name: "l7-ilb-regional-url-map"
+          backend_service_name: "l7-ilb-backend-subnet"
+          mig_template_name: "l7-ilb-mig-template"
+          hc_name: "l7-ilb-hc"
+          mig_name: "l7-ilb-mig1"
+          fw-allow_iap_hc_name: "l7-ilb-fw-allow_iap_hc"
+          fw_allow_ilb_to_backends_name: "l7-ilb-fw_allow_ilb_to_backends"
```
Comment (c2thorn) [inline mmv1/products/compute/terraform.yaml:720]: Let's swap the underscores for hyphens for these. I ran into a validation error when underscores were used for the name

Comment (betsy-lichtenberg) [reply mmv1/products/compute/terraform.yaml]: done

File: mmv1/templates/terraform/examples/internal_http_lb_with_mig_backend.tf.erb [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+  }
+}
+
+# allow http from proxy subnet to backends
+resource "google_compute_firewall" "fw-ilb-to-backends" {
+  name          = "<%= ctx[:vars]['fw_allow_ilb_to_backends_name'] %>"
+  provider      = google-beta
+  direction     = "INGRESS"
+  network       = google_compute_network.ilb_network.id
+  source_ranges = "10.0.0.0/24"
```
[the remark this answers is not in the document]
Comment (c2thorn) [reply mmv1/templates/terraform/examples/internal_http_lb_with_mig_backend.tf.erb:167]: This needs to be an array of strings I believe

**Review by c2thorn — CHANGES_REQUESTED** [review]

**Review by c2thorn — APPROVED** [review]
@betsy-lichtenberg I've tested your example locally, and the test passed! I can't merge just yet because we need the test to succeed in our build system which has some bugs we are working through at the moment. When that is back up, I can merge.

**Comment by c2thorn** [comment]
Failing tests were unrelated

**Pull request merged**